packages/echo-base/src/errors/BaseError.test.ts

```diff
@@ -0,0 +1,68 @@
+import { BaseError, getAllProperties } from './BaseError';
+
+describe('BaseError', () => {
+    const message = 'this is a test';
+    it('should initialize with correct values', () => {
+        const actualError = new BaseError({ name: 'BaseError', message });
+        expect(actualError.message).toBe(message);
+        expect(actualError.name).toBe('BaseError');
+        expect(actualError.stack).toBeTruthy();
+    });
+
+    it('should preserve innerError as properties', () => {
```

Comment:
Hmmm, I was confused here by this 'innerError' a bit... 🤔 

Maybe some re-namings would help to get the full picture a bit more easly?
```ts
it('should preserve a given custom error as a property', () => {
  const customErrorMessage = 'Custom error message';
  const customException = new CustomError('Custom error message');
  const actualError = new BaseError({ name: 'BaseError', message, exception: customException });
  ...
});
```